hledger-lib/Hledger/Reports/ReportOptions.hs

```diff
@@ -118,6 +118,7 @@ data ReportOpts = ReportOpts {
     ,drop_           :: Int
     ,row_total_      :: Bool
     ,no_total_       :: Bool
+    ,show_costs_     :: Bool  -- ^ Whether to show costs for reports which normally don't show them
```

Comment:
I haven't got my head around the larger effect of this, so I'll just ask questions. Will --show-costs now be understood by more commands ? Which ones ? Will/should it appear in more commands' help ? And in the manual ? Will some commands show it in help but ignore it ? Will it cause bad effects with any commands ?

Reply:
As implemented here, it will not be understood by more commands: it will still only be understood by the close command. The only difference at this stage is that the corresponding Bool has been moved into `ReportOpts`, rather than living in `Close.hs`. This change obviously suggests that the flag could be understood more widely, but it would require some more changes to work properly. That's probably an issue for another PR, if it's even decided it's worthwhile at all (I'm sceptical at this stage).